__tests__/CollapsedMetrics.test.js

```diff
@@ -0,0 +1,611 @@
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.833368]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 38, 1.821198]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 39, 1.08]
+                },
+            ]
+        }
+
+        let result = sut.getAvgofAvgVelocities(set);
+
+        expect(result).toBe("1.58");
+    });
+});
+
+describe('getAbsLossVelocity', () => {
+
+    test('return null when avgVs empty', () => {
+        let set = {
+            reps: [],
+        };
+        
+        let result = sut.getAbsLossVelocity(set);
+        
+        expect(result).toBe(null);        
+    });
+
+    test('returns abslossvelocity', () => {
+        let set = {
+            reps: [
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.833368]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 38, 1.821198]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 39, 1.08]
+                },
+            ]
+        }
+        
+        let result = sut.getAbsLossVelocity(set);
+
+        expect(result).toBe(0.75);
+    });
+});
+
+describe('getFirstRepAvgVelocity', () => {
+    
+    test('return null when avgVs empty', () => {
+        let set = {
+            reps: [],
+        };
+            
+        let result = sut.getFirstRepAvgVelocity(set);
+            
+        expect(result).toBe(null);        
+    });
+    
+    test('returns first rep avg velocity', () => {
+        let set = {
+            reps: [
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 37, 1.833368]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 38, 1.821198]
+                }, 
+                {
+                    isValid: true,
+                    removed: false,
+                    data: [-3456, 39, 1.08]
+                },
+            ]
+        }
+            
+        let result = sut.getFirstRepAvgVelocity(set);
+    
+        expect(result).toBe(1.83);
+    });
+});
+    
+describe('getLastRepAvgVelocity', () => {
+    
+    test('return null when avgVs empty', () => {
+        let set = {
+            reps: [],
+        };
+            
+        let result = sut.getLastRepAvgVelocity(set);
+            
+        expect(result).toBe(null);        
+    });
+    
+    test('returns last rep avg velocity', () => {
+        let set = {
```

Comment:
- [ ]   modify test to have a invalid and removed values for the last 2 reps that aren't used